tests/receivers/smartagent/haproxy/haproxy_test.go

```diff
@@ -17,13 +17,125 @@
 package tests
 
 import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
 	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+	"go.uber.org/zap"
 
 	"github.com/signalfx/splunk-otel-collector/tests/testutils"
 )
 
 func TestHaproxyReceiverProvidesAllMetrics(t *testing.T) {
-	testutils.AssertAllMetricsReceived(
-		t, "all.yaml", "all_metrics_config.yaml", nil, nil,
+	checkGoldenFile(t, "all_metrics_config.yaml", "all_expected.yaml",
+		pmetrictest.IgnoreMetricDataPointsOrder(),
+		pmetrictest.IgnoreTimestamp(),
+		pmetrictest.IgnoreStartTimestamp(),
+		pmetrictest.IgnoreMetricAttributeValue("unique_proxy_id",
+			"haproxy_bytes_out",
+			"haproxy_compress_bypass",
+			"haproxy_compress_in",
+			"haproxy_compress_out",
+			"haproxy_compress_responses",
+			"haproxy_denied_request",
+			"haproxy_denied_response",
+			"haproxy_intercepted_requests",
+			"haproxy_connection_rate",
+			"haproxy_connection_rate_max",
+			"haproxy_connection_total",
+			"haproxy_request_rate_max",
+			"haproxy_error_connections",
+			"haproxy_retries",
+			"haproxy_bytes_in",
+			"haproxy_bytes_out",
+			"haproxy_compress_bypass",
+			"haproxy_active_servers",
+			"haproxy_backup_servers",
+			"haproxy_client_aborts",
+			"haproxy_redispatched",
+			"haproxy_request_total",
+			"haproxy_response_1xx",
+			"haproxy_response_2xx",
+			"haproxy_response_3xx",
+			"haproxy_response_4xx",
+			"haproxy_response_5xx",
+			"haproxy_response_other",
+			"haproxy_response_time_average",
+			"haproxy_server_aborts",
+			"haproxy_server_selected_total",
+			"haproxy_session_current",
+			"haproxy_session_limit",
+			"haproxy_session_max",
+			"haproxy_session_rate",
+			"haproxy_session_rate_max",
+			"haproxy_session_time_average",
+			"haproxy_session_total",
+			"haproxy_status",
+		),
+		pmetrictest.IgnoreMetricsOrder(),
+		pmetrictest.IgnoreMetricDataPointsOrder(),
+		pmetrictest.IgnoreMetricValues(
+			"haproxy_intercepted_requests",
+			"haproxy_bytes_in",
+			"haproxy_bytes_out",
+			"haproxy_connection_total",
+			"haproxy_request_total",
+			"haproxy_response_2xx",
+			"haproxy_session_current",
+			"haproxy_session_total",
+		),
 	)
 }
+
+func checkGoldenFile(t *testing.T, configFile string, expectedFilePath string, options ...pmetrictest.CompareMetricsOption) {
```

Comment:
Would it make sense to move this function to some central location that can be reused? It looks like it's being defined for every component, but maybe there are differences I'm missing.

Reply:
yes, of course. I wanted my PRs to be independent, but we can refactor.